### PR TITLE
podman-remote prune containers

### PR DESCRIPTION
**Pull request by baude**
enable the ability to prune containers from the remote-command.  this
also includes the system prune command.

Signed-off-by: baude <bbaude@redhat.com>

### CODE REVIEW

**Comment by rhatdan** [comment]
Needs a rebase to get the SELinux fixes.

File: test/e2e/libpod_suite_test.go [hunk 1]
```diff
@@ -13,7 +13,6 @@ import (
 )
 
 func SkipIfRemote() {
-	ginkgo.Skip("This function is not enabled for remote podman")
```
Comment (rhatdan) [inline test/e2e/libpod_suite_test.go]: Why are you deleting this line?

File: pkg/adapter/containers.go [hunk 2]
```diff
@@ -694,3 +694,44 @@ func (r *LocalRuntime) UnpauseContainers(ctx context.Context, cli *cliconfig.Unp
...
+		ok       = []string{}
+		failures = map[string]error{}
+		err      error
+	)
+
+	logrus.Debugf("Setting maximum rm workers to %d", maxWorkers)
+
+	filter := func(c *libpod.Container) bool {
+		state, err := c.State()
+		if state == libpod.ContainerStateStopped || (state == libpod.ContainerStateExited && err == nil && c.PodID() == "") {
```
Comment (mheon) [inline pkg/adapter/containers.go]: The logic here seems screwy.
Why are stopped and exited different? Why are we checking err (it's guaranteed that if err != nil, we never return a sensible state, only CtrStateUnknown)

**Comment by rhatdan** [comment]
@baude Needs a rebase.

File: test/e2e/libpod_suite_test.go (hunk 1 above)
Comment (baude) [reply test/e2e/libpod_suite_test.go]: because skipifremote should not do anything when testing "locally"

File: pkg/adapter/containers.go (hunk 2 above)
Comment (baude) [reply pkg/adapter/containers.go]: this is exactly what was in cmd/podman/containers_prune.go ... I changed nothing.  You'd have to ask the original author.

Comment (mheon) [reply pkg/adapter/containers.go]: I'm gonna propose we make the PodID check separate (if `podID != ""` always return false) and remove the err == nil check.

**Comment by jwhonce** [comment]
LGTM

**Pull request merged**